gslib/utils/shim_util.py

```diff
@@ -401,7 +403,10 @@ def _translate_top_level_flags(self):
       top_level_flags.append('--impersonate-service-account=' +
                              constants.IMPERSONATE_SERVICE_ACCOUNT)
     # TODO(b/208294509) Add --perf-trace-token translation.
-    if not self.parallel_operations:
+    rsync_parallelism_disabled = self.command_name != 'rsync' or (
```

Comment:
With rsync in the variable name it's a bit counterintuitive for this to be true when the command is not rsync. How about something like this?
 
```
should_use_rsync_override = self.command_name == 'rsync' and not (process and thread count check)
if not self.parallel_operations and not should_use_rsync_override:
  ...
```

Reply:
Good idea